api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -277,22 +284,24 @@ public String getName() {
 	public void setName(String name) {
 		this.name = name;
 	}
-	
+
+	@Deprecated
 	public Set<Integer> getMemberIds() {
 		return memberIds;
 	}
-	
+
+	@Deprecated
 	public void setMemberIds(Set<Integer> memberIds) {
 		this.memberIds = new TreeSet<Integer>(memberIds);
 	}
 
-    public Set<Integer> getMembers() {
-        return members;
-    }
+	public Collection<CohortMember> getMembers() {
+		return members;
+	}
 
-    public void setMembers(Set<Integer> members) {
```

Comment:
Same here, deprecate the old setter and keep them around otherwise this is going to break external code